src/elements/content-sidebar/activity-feed/task-form/TaskForm.js

```diff
@@ -190,14 +215,21 @@ class TaskForm extends React.Component<Props, State> {
     };
 
     handleApproverSelectorSelect = (pills: any): void => {
-        this.setState({ approvers: this.state.approvers.concat(pills) });
+        // TODO: remove duplicates in added list
+        this.setState({ approvers: this.state.approvers.concat(pills), addedAssignees: pills });
         this.validateForm('taskAssignees');
     };
 
     handleApproverSelectorRemove = (option: any, index: number): void => {
         const approvers = [...this.state.approvers];
-        approvers.splice(index, 1);
-        this.setState({ approvers });
+        const fullApprovers = this.props.approvers;
```

Comment:
We should be using destructure assignments whenever possible, even if it means destructing 1. Helps in diffing if in the future more things need to get added. Similar to trialing , logic. And prevents multiple this.state.*

const { approvers, removedAssignees } = this.state;
const { approvers: fullApprovers } = this.props;
const approversClone/copy = [...approvers];

Reply:
a lot of this got factored out so resolving the comments in this block